File: day1-2.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args){
        String filename = "/Users/shun/Code/small-tasks/adventofcode2018/day1.txt";
        try (BufferedReader reader = new BufferedReader(new FileReader(new File(filename)))){
            reader.mark(8196);// 最初にマーク
            String line;
            Integer answer = 0;
            List<Integer> answers = new ArrayList<Integer>();
            Boolean flag = false;
            Integer n = 0;
            while(flag == false) {
              while((line = reader.readLine()) != null) {
                //System.out.println(line);
                answer += Integer.parseInt(line);
                if(answers.contains(answer)) {
                  System.out.println(answer);
                  flag = true;
                  break;
                }
                n += 1;
                answers.add(answer);
              }
              reader.reset();
            }
        } catch (FileNotFoundException e){ 
            e.printStackTrace();
            System.exit(-1);
        } catch (IOException e){ 
            e.printStackTrace();
            System.exit(-1);
        }
    }   
}

File: day1-2.rb
File.open("day1.txt") do |f|
  input = []
  f.each_line.with_index do |line, idx|
    num = line.to_i
    input[idx] = num
  end

  flag = false
  current = 0
  answers = []
  until flag == true
    input.each do |num|
      current += num
      if answers.include?(current)
        flag = true
        p current
        break
      else 
        answers << current
      end
    end
  end
end